utils/react/ReactUtils.js

```diff
@@ -5,6 +5,40 @@
  * Copyright © 2020 Extremely Heavy Industries Inc.
  */
 
+import {isValidElement, createElement, cloneElement} from 'react';
+import {throwIf} from '../js';
+
+/**
+ * Return the display name for either a class-based or functional Component.
+ * @param obj
+ * @return {string}
+ */
 export function getReactElementName(obj) {
-    return obj.type.name || obj.type.displayName;  // Support for class-based and functional cmps, respectively
+    return obj.type.name || obj.type.displayName;
+}
+
+/**
+ * Create a React element from either a HoistComponent or a function returning an element.
+ *
+ * Used by the TabContainer, DashContainer, and DockView APIs to process the 'content' configs
+ * provided to them for their tabs and views.
+ *
+ * @param {(Object|function)} content - HoistComponent or function returning a React element. If a
+ *      function, it may be an 'elemFactory' (as created by elemFactory()), or any function that
+ *      returns a React element. In either case, the function will be called with no arguments.
+ * @param {Object} [addProps] -- optional additional props to apply to the element.
+ *      These will override any existing props placed on the element, and should be used with care.
+ */
+export function elementFromContent(content, addProps) {
+
+    // Note: Would be more general to look for a *react* component.
```

Comment:
Should we do this now? 😃

Reply:
its actually surprisingly tricky with functional components and class components.  We have a clear way to identify a HoistComponent, and that's the way the code currently works.

I guess I just put the comment there because I felt guilty.  :-)